packages/hash/subgraph/tests/compatibility.test.ts

```diff
@@ -0,0 +1,30 @@
+import { Subgraph as SubgraphGraphApi } from "@hashintel/hash-graph-client";
+import { Subgraph } from "../src";
+import { mapVertices } from "./compatibility.test/map-vertices";
+import { mapEdges } from "./compatibility.test/map-edges";
+
+jest.setTimeout(60000);
+
+test("Graph API subgraph type is compatible with library type", () => {
+  // We don't need an actual subgraph, we are just checking for TSC errors
+  const subgraphGraphApi: SubgraphGraphApi = {
+    roots: [],
+    vertices: {},
+    edges: {},
+    depths: {
+      dataTypeResolveDepth: 0,
+      propertyTypeResolveDepth: 0,
+      entityTypeResolveDepth: 0,
+      entityResolveDepth: 0,
+    },
+  };
+
+  // We just want to check for errors in the type when building the object, no need to use the return value
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const subgraph: Subgraph = {
```

Comment:
I think could also:
```suggestion
  const _subgraph: Subgraph = {
```

Reply:
So I can. I tried `const _ = {` and it got annoyed, but hadn't tried the prefix.